File: website/app/Models/Job.php
<?php

namespace App\Models;

use App\User;
use Eloquent as Model;
use Illuminate\Database\Eloquent\SoftDeletes;

/**
 * Class Job
 * @package App\Models
 * @version March 12, 2020, 5:43 pm UTC
 *
 * @property string titre
 * @property string typeoffre
 * @property string secteuractivite
 * @property string niveauetude
 * @property string lieu
 * @property string datelimite
 * @property string description
 * @property integer nombreposte
 */
class Job extends Model
{
    use SoftDeletes;

    public $table = 'jobs';


    protected $dates = ['deleted_at'];



    public $fillable = [
        'titre',
        'typeoffre',
        'secteuractivite',
        'niveauetude',
        'lieu',
        'datelimite',
        'description',
        'nombreposte'
    ];

    /**
     * The attributes that should be casted to native types.
     *
     * @var array
     */
    protected $casts = [
        'id' => 'integer',
        'titre' => 'string',
        'typeoffre' => 'string',
        'secteuractivite' => 'string',
        'niveauetude' => 'string',
        'lieu' => 'string',
        'description' => 'string',
        'datelimite' => 'date',
        'nombreposte' => 'integer'
    ];

    /**
     * Validation rules
     *
     * @var array
     */
    public static $rules = [

    ];
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function postuler(){
        return $this->hasMany(Postuler::class);
    }

}

File: mobile/emploici/lib/Data/Model/Job.dart

import 'package:emploici/Data/Model/User.dart';

class Job  {
  int id;
  String titre;
  String typeoffre;
  String secteuractivite;
  String niveauetude;
  String lieu;
  String datelimite;
  String description;
  int nombreposte;
  User user;

  Job({this.id, this.titre, this.typeoffre,
    this.secteuractivite, this.niveauetude, this.lieu, this.datelimite, this.description, this.nombreposte, this.user});



  Map<String, dynamic> toJson() => <String, dynamic>{
    'id': this.id,
    'titre': this.titre,
    'typeoffre': this.typeoffre,
    'secteuractivite': this.secteuractivite,
    'niveauetude': this.niveauetude,
    'lieu': this.lieu,
    'datelimite': this.datelimite,
    'description': this.description,
    'nombreposte': this.nombreposte,
    'user': this.user,
  };

  factory Job.fromJson(Map<String, dynamic> json) {
    return Job(
      id: json['id'] as int,
      titre: json['titre'] as String,
      typeoffre: json['typeoffre'] as String,
      secteuractivite: json['secteuractivite'] as String,
      niveauetude: json['niveauetude'] as String,
      lieu: json['lieu'] as String,
      datelimite: json['datelimite'] as String,
      description: json['description'] as String,
      nombreposte: json['nombreposte'] as int,
      user: json['user'] == null
          ? null
          : User.fromJson(json['user'] as Map<String, dynamic>),
    );
  }



}